tests/data/mini_c_test/b_test.cc

```diff
@@ -17,42 +17,54 @@ TEST(b_test, TestSignalChain_1)
 
 TEST(b_test, TestSignalChain_2)
 {
-  CREATE_MOCK(mymock);
+  LOCAL_MOCK(mymock);

```

Comment:
Just to clarify if I understood it correctly: this is a breaking change, right? I mean if I have tests in my projects using CREATE_MOCK() inside my TESTs.

Reply:
> Just to clarify if I understood it correctly: this is a breaking change, right? I mean if I have tests in my projects using CREATE_MOCK() inside my TESTs.

That's right. We need a mock that is destroyed automatically, and another one that is destroyed explicitly, to be assigned to - for example - a class property.
When "CREATE_MOCK" is already the temporary version, I have no idea for the persistent version.

Reply:
> And another question: with hammocking 0.2.5 we are wasting memory as CREATE_MOCK(name) creates a new mock everytime and we never cleanup?
No; it is cleaned up when the local variable gets out of scope.
Mechanism:
`CREATE_MOCK(m)` creates the local object `m` as instance of `class_mockup`. Its constructor assigns the object also to the global `mockup_global_ptr` (which is needed to redirect the C function call).
When `m` now gets out of scope, it gets destroyed. Expectations are finalized; the destructor also resets the `mockup_global_ptr`.

New usecase:
You want to preload some common behavior in the `SetUp()` method. A local mock would be destroyed with the end of `SetUp()` (which will also delete the preloaded behavior), so the registry now needs to survive until the test function is over. There is no scope that expands this lifespan, so the only way is to create a mock in `SetUp()`, store it somewhere (e.g. a fixture property) and destroy is explicitly in `TearDown()`. 
This mock is persistent since it is created with `new`, and lives until destroyed with `delete`.
There is such a thing as `shared_ptr` in C++, which has the task of making deleting the object foolproof by keeping track by itself if someone still has a copy of the pointer. However, I didn't get it to work. It also does not solve the problem that something needs to be placed in `TearDown()` to unreference the registry.

Reply:
Now I remember: The idea with the `shared_ptr` was that when you re-assign the `mockup_global_ptr` in the following test, the old object gets unused and would be deleted.
But gmock didn't like it when the registry survived the test function teardown, and complained heavily. The same for when it still existed at test run end.

Reply:
Okay, all understood. Proposal for the naming to stay backward compatible: CREATE_MOCK(name) stays the way it was.

Naming proposal for macros handling the persistent mockup creation:
CREATE_PERSISTENT_MOCK()
DESTROY_PERSISTENT_MOCK()

Reply:
`Backward compatible`, hah... The change is in the repository for 1 year now, and productively used by myself for 1 1/2 years, so not doing it like proposed is a breaking change for _me_.
But that's it when you use unreleased stuff...

Reply:
Yes, we are using the latest release since over a year. So being backward compatible to the latest release sounds nice.